### PR TITLE
LPD-33284 | LPD-33599

### PR DESCRIPTION
**Pull request by cdbm**
(pr for review while master syncs, consider only last two commits) 
- [LPD-33284](https://liferay.atlassian.net/browse/LPD-33284 ) 
- [LPD-33599](https://liferay.atlassian.net/browse/LPD-33599)

### CODE REVIEW

**Comment by cdbm** [comment]
ci:test:sf

File: modules/dxp/apps/scim/scim-rest-impl/src/main/java/com/liferay/scim/rest/internal/configuration/ScimClientOAuth2ApplicationConfigurationFactory.java [hunk 1]
```diff
@@ -111,11 +112,10 @@ protected void deactivate(Integer reason) throws PortalException {
...
 			long companyId,
 			ScimClientOAuth2ApplicationConfiguration
-				scimClientOAuth2ApplicationConfiguration)
+				scimClientOAuth2ApplicationConfiguration,
+			long userId)
 		throws Exception {
 
-		User user = _userLocalService.getGuestUser(companyId);
-
 		User clientCredentialUser = _userLocalService.getUserByScreenName(
```
Comment (ChrisKian) [inline modules/dxp/apps/scim/scim-rest-impl/src/main/java/com/liferay/scim/rest/internal/configuration/ScimClientOAuth2ApplicationConfigurationFactory.java:119]: Do we also need to change this `clientCredentialUser` to the current user?  Could you please try adjusting this and see if everything works as intended?

File: modules/dxp/apps/scim/scim-rest-impl/src/main/java/com/liferay/scim/rest/internal/configuration/ScimClientOAuth2ApplicationConfigurationFactory.java [hunk 2]
```diff
@@ -72,7 +72,8 @@ protected void activate(
 							properties);
 
 				_oAuth2Application = _getOrAddOAuth2Application(
-					companyId, scimClientOAuth2ApplicationConfiguration);
+					companyId, scimClientOAuth2ApplicationConfiguration,
+					(long)properties.get("userId"));
```
Comment (ChrisKian) [inline modules/dxp/apps/scim/scim-rest-impl/src/main/java/com/liferay/scim/rest/internal/configuration/ScimClientOAuth2ApplicationConfigurationFactory.java:76]: Please change this to `GetterUtil.getLong(properties.get("userId")));` and adjust SF if needed

File: modules/test/playwright/tests/scim-configuration-web/scim.spec.ts [hunk 3]
```diff
@@ -229,3 +229,78 @@ test('LPD-23255 AC5 TC8: Verify that the Name field is enabled when scim client
...
+		},
+		userName: `able${randomNumber}.baker`,
+	};
+
+	const apiHelper = new ApiHelpers(page);
+
+	await apiHelper.scim.postUser(newUser, accessToken);
+
+	const response = await (await apiHelper.scim.getUsers(accessToken)).text();
+	console.log(response);
```
Comment (ChrisKian) [inline modules/test/playwright/tests/scim-configuration-web/scim.spec.ts:270]: You forgot to take this out

**Comment by cdbm** [comment]
reopened here https://github.com/liferay-appsec/liferay-portal/pull/1907

**Pull request closed without merging**